embedding-stores/weaviate/deployment/src/test/java/io/quarkiverse/langchain4j/weaviate/deployment/WeaviateEmbeddingStoreTest.java

```diff
@@ -0,0 +1,104 @@
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import dev.langchain4j.store.embedding.weaviate.WeaviateEmbeddingStore;
+import io.quarkus.test.QuarkusUnitTest;
+import io.weaviate.client.Config;
+import io.weaviate.client.WeaviateClient;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.v1.data.model.WeaviateObject;
+
+class WeaviateEmbeddingStoreTest extends EmbeddingStoreIT {
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(
+                            new StringAsset(String.format("quarkus.langchain4j.weaviate.metadata.keys=%s",
+                                    Constants.METADATA_KEYS.stream().collect(Collectors.joining(",")))),
+                            "application.properties"));
+
+    @ConfigProperty(name = "quarkus.langchain4j.weaviate.scheme")
+    String weaviateScheme;
+
+    @ConfigProperty(name = "quarkus.langchain4j.weaviate.host")
+    String weaviateHost;
+
+    @ConfigProperty(name = "quarkus.langchain4j.weaviate.port")
+    Integer weaviatePort;
+
+    @ConfigProperty(name = "quarkus.langchain4j.weaviate.metadata.keys")
+    List<String> weaviateMetadataKeys;
+
+    String weaviateClassName = "Default";
+
+    private WeaviateEmbeddingStore embeddingStore;
+    private static EmbeddingModel embeddingModel;
+
+    /**
+     * FIXME: This is a workaround to avoid loading the embedding model in this test class' static initializer,
+     * because otherwise we hit
+     * java.lang.UnsatisfiedLinkError: Native Library (/path/to/the/library) already loaded in another classloader
+     * because the test class is loaded by JUnit and by Quarkus in different class loaders.
+     */
+    @BeforeAll
+    public static void initEmbeddingModel() {
+        embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
+    }
+
+    protected WeaviateEmbeddingStore embeddingStore() {
+        if (embeddingStore == null) {
+            embeddingStore = WeaviateEmbeddingStore.builder()
+                    .scheme(weaviateScheme)
+                    .host(weaviateHost)
+                    .port(weaviatePort)
+                    .objectClass(weaviateClassName)
+                    .metadataKeys(weaviateMetadataKeys)
+                    .build();
+        }
+        return embeddingStore;
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        this.ensureStoreIsReady();
+        this.clearStore();
+        this.ensureStoreIsEmpty();
+    }
+
+    protected EmbeddingModel embeddingModel() {
+        return embeddingModel;
+    }
+
+    protected void ensureStoreIsReady() {
+
+    }
+
+    protected void ensureStoreIsEmpty() {
+    }
+
+    protected void clearStore() {
+        WeaviateClient client = new WeaviateClient(new Config(weaviateScheme, weaviateHost + ":" + weaviatePort));
```

Comment:
Do we know what the client uses? HTTP, gRPC? At some point we probably want to use our own support for these stacks as it gets us stuff like context propagation, metrics, traces etc. for free

Reply:
It uses http and optionally grpc for batch inserts.

Reply:
I see. At some point it would be great to use our HTTP stack